github/create_deployment.yaml

```diff
@@ -0,0 +1,109 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: github-create-deployment
+  description: |
+    This Task will create a GitHub deployment for a repository.
+spec:
+  params:
+    - name: GITHUB_HOST_URL
+      description: |
+        The GitHub host, adjust this if you run a GitHub enterprise.
+      default: "api.github.com"
+    - name: REPO_FULL_NAME
+      description: |
+        The GitHub repository full name, i.e: tektoncd/catalog.
+    - name: REF
+      description: |
+        The ref to deploy. This can be a branch, tag, or SHA.
+    - name: TASK
+      description: |
+        Specifies a task to execute (e.g., deploy or deploy:migrations).
+      default: "deploy"
+    - name: AUTO_MERGE
+      description: |
+        Attempts to automatically merge the default branch into the
+        requested ref, if it's behind the default branch.
+      default: "true"
```

Comment:
Would be great if boolean was a supported type so that we don't have to do a `json.loads()` further below to coerce this into a boolean that the API expects. This is my workaround for now.